pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -396,96 +396,115 @@ static PyObject *get_item(PyObject *obj, Py_ssize_t i) {
     return ret;
 }
 
-static void *PyBytesToUTF8(JSOBJ _obj, JSONTypeContext *tc, void *outValue,
-                           size_t *_outLen) {
+static char *PyBytesToUTF8(JSOBJ _obj, JSONTypeContext *tc, size_t *_outLen) {
     PyObject *obj = (PyObject *)_obj;
     *_outLen = PyBytes_GET_SIZE(obj);
     return PyBytes_AS_STRING(obj);
 }
 
-static void *PyUnicodeToUTF8(JSOBJ _obj, JSONTypeContext *tc, void *outValue,
-                             size_t *_outLen) {
-  return PyUnicode_AsUTF8AndSize(_obj, _outLen);
+static char *PyUnicodeToUTF8(JSOBJ _obj, JSONTypeContext *tc, size_t *_outLen) {
+    return (char *)PyUnicode_AsUTF8AndSize(_obj, _outLen);
 }
 
-static void *PandasDateTimeStructToJSON(npy_datetimestruct *dts,
```

Comment:
The diff is a little difficult to read here, but this got rid of the `ToJSON` functions and instead replaces them with explicit `ToIso` or `ToEpoch` functions with strong types